crates/kumod/src/ready_queue.rs

```diff
@@ -729,18 +727,26 @@ impl Drop for Dispatcher {
                         } else {
                             if let Err(err) = Dispatcher::requeue_message(msg, false, None).await {
                                 tracing::error!("error requeuing message: {err:#}");
-                            } else {
-                                tokio::time::sleep(Duration::from_secs(1)).await;
-                                let ready_queue = ReadyQueueManager::get_by_name(&name);
-                                if let Some(q) = ready_queue {
-                                    q.notify_maintainer.notify_one();
-                                }
-                                notify_dispatcher.notify_one();
                             }
                         }
                     })
                 })
                 .ok();
+        } else {
```

Comment:
Ah, I see what you're saying. I think that the shape of the logic in here could be tweaked a bit to look roughly like this:

```rust
fn drop(&mut self) {
   let msg = self.msg.take();
   READYQ_RUNTIME
                .spawn_non_blocking("Dispatcher::drop".to_string(), move || {
             Ok(async move {   
                 if let Some(msg) = msg {
                       if activity.is_shutting_down() {
                            Queue::save_if_needed_and_log(&msg).await;
                       } else {
                            if let Err(err) = Dispatcher::requeue_message(msg, false, None).await {
                                tracing::error!("error requeuing message: {err:#}");
                             }
                       }
                 } else {
                        tokio::time::sleep(Duration::from_secs(1)).await;
                        let ready_queue = ReadyQueueManager::get_by_name(&name);
                        if let Some(q) = ready_queue {
                            q.notify_maintainer.notify_one();
                        }
                        notify_dispatcher.notify_one();
                  }
          }
}

Reply:
I pushed an update, it looks better than my old copy paste :)